pages/business/company/[companyId]/checkins.tsx

```diff
@@ -29,45 +32,68 @@ const CheckinsPage: React.FC<WithOwnerProps> = () => {
       <Box height={2} />
       <ActionList grid>
         {company?.areas.map((area) => {
-          const tickets = ticketsByArea?.[area.id]
-          const openCount = tickets?.open.length ?? 0
-          const closedCount = tickets?.closed.length ?? 0
-
-          return (
-            <ActionCard
-              key={area.id}
-              href="/business/company/[companyId]/area/[areaId]"
-              as={`/business/company/${companyId}/area/${area.id}`}
-            >
-              <ActionCard.Main
-                title={area.name}
-                subtitle={
-                  <>
-                    <span css={{ whiteSpace: 'nowrap' }}>
-                      am Tisch: {openCount}
-                    </span>
-                    {' – '}
-                    <span css={{ whiteSpace: 'nowrap' }}>
-                      ausgecheckt: {closedCount}
-                    </span>
-                  </>
-                }
-              />
-              <ActionCard.Below>
-                <Row flexWrap="wrap" flex="1 0">
-                  {[...Array(openCount)].map((_v, i) => (
-                    <Box key={i} mx="2px">
-                      <Icon icon={TinyCheck} size={4} />
-                    </Box>
-                  ))}
-                </Row>
-              </ActionCard.Below>
-            </ActionCard>
-          )
+          return <CheckinsActionCard area={area} companyId={companyId} />
         })}
       </ActionList>
     </OwnerApp>
   )
 }
 
+const CheckinsActionCard: React.FC<{
+  area: AreaRes
+  companyId: string
+}> = ({ area, companyId }) => {
+  const { data: ticketsByArea } = useLastTicketsGrouped(companyId)
+
+  const tickets = ticketsByArea?.[area.id]
+  const openCount = tickets?.open.length ?? 0
+  const closedCount = tickets?.closed.length ?? 0
+  const [isLoading, setLoading] = React.useState(false)
+
+  const checkoutAll = async () => {
+    setLoading(true)
+
+    const leftAt = new Date()
+
+    await tickets?.open.map(({ id }) => patchTicket({ id, leftAt }))
```

Comment:
This doesn't work – at least it doesn't do what you intend to do. This `await` resolves immediately with an array of _pending promises_. It does not wait for the promises themselves to resolve. It would do the same if you would remove the `await`.

If you want to wait until an array of promises is resolved, you need to use `Promise.all(array)`

```suggestion
    const openTickets = tickets?.open || []
    await Promise.all(openTickets.map(({ id }) => patchTicket({ id, leftAt })))
```

Reply:
Damn, good catch! Thanks :)